libraries/Microsoft.Bot.Builder.Dialogs/Memory/Scopes/LGMemoryScope.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.Builder.Dialogs.Memory.Scopes
+{
+    /// <summary>
+    /// DialogMemoryScope maps "lg" -> dc.Parent?.ActiveDialog.State ?? ActiveDialog.State.
```

Comment:
I think creating a new scope is not just for a different prefix, but rather creating a new temp space to hold the states like turn scope.